_posts/2023-09-10-python-multiple-arithmetic-operators.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "The surprising behavior of multiple + and - operators in Python"
+header:
+  image: /assets/images/headers/2023-09-28_pythons-weird-arithmetic.png
+  image_description: "An example of Python's weird arithmetic"
+  teaser: /assets/images/teasers/2023-09-28_pythons-weird-arithmetic.png
+excerpt: Let's explore this strange quirk of the Python language and learn from it.
+---
+
+All programming languages have some quirks and strange behavior.
+Python is certainly no exception.
+I was however quite surprised recently when I found out that Python allows multiple `+` and `-` in a row.
+Let's explore this behavior and learn about order of operation from it.
+
+# Use as many `+` and `-` together as you want
+
+I do not know the purpose of this feature but in Python you can write multiple `+` or `-` operators in a row.
+
+```python
+>>> 1 +++++++ 1
+2
+>>> 1 ++++++++++++++++++++ 1
+2
+>>> 1 ------- 1
+0
+>>> 1 ------ 1
+2
+```
+
+The operators behave as if they were shorthands for `* (+1)` and `* (-1)`.
+I do not know of any other language that behaves like this.
+I also do not see any purpose to it.
+
+The only use case that I can think of is if you need to ask "help" from a fellow developer to deal with your annoying manager:
+
+```python
+my_need = 1 --- +++ --- 5
+```
+
+There is always someone familiar with morse code in movies so it might work.
+
+# Using multiple `-` with `//` is weird
+
+In case you don't know, the `//` operator performs a division like `/` but instead returns an integer.
+When writing `5 // 3` you are doing the exact same calculation as `math.floor(5 / 3)`.
+
+```python
+>>> 5 // 3
+1
+```
+
+Now let's look at this simple calculation using `//`:
+
+```python
+>>> 2 - 5 // 2
+0
+```
+
+The result is as expected since `5 // 2` is equivalent to `floor(5 / 2)` which is equal to 2.
+
+According to the previous section, if we use `--` instead of `-`, we should expect a result of 4.
+However here is what we get:
```

Comment:
Maybe more emotional language? Again depends on audience goals. Shockingly, surprisingly, broke my brain...

Edit: ah you dropped the emotion below! Nice. Maybe we cut this - or at least remove the language that gives away the surprise. It's still a mystery right now!

I wonder what will happen if... milk the storyline!